lib/ecto/migration.ex

```diff
@@ -921,11 +926,12 @@ defmodule Ecto.Migration do
 
   @doc false
   def __prefix__(%{prefix: prefix} = index_or_table) do
+    default_prefix = Runner.repo_config(:migration_default_prefix, nil)
     runner_prefix = Runner.prefix()
```

Comment:
I wonder if the `Runner.prefix()` call should handle this logic internally.

Reply:
Sounds like a good candidate, yeah!

Reply:
I had the runner handle it internally, but then the exception condition below raised. To make it handle overriding and not raising an exception on the match I ended up checking to see if the `migration_default_prefix` was set or not, and at that point I figured since I already had to pull in the value, that this approach had the same effect with less code.

Reply:
@coryodaniel why did the exception raised though? Why were both being given?

Reply:
It raised when it was internal to `Runner.prefix()` it fails the following scenario, which I think we should allow to be overridden.

```elixir
@tag repo_config: [migration_default_prefix: "baz"]
  test "forward: creates a table with prefix from migration overriding the default prefix configuration" do
    create(table(:posts, prefix: "foo"))
    flush()

    {_, table, _} = last_command()
    assert table.prefix == "foo"
  end
```